middle_end/flambda/inlining/inlining_report.ml

```diff
@@ -136,12 +142,22 @@ let record_decision ~dbg decision =
   end
 
 let output_then_forget_decisions ~output_prefix =
+  let l = lazy (List.rev !log) in
   if !Clflags.inlining_report then begin
     let out_channel = open_out (output_prefix ^ ".inlining.org") in
     let fmt = Format.formatter_of_out_channel out_channel in
-    Format.fprintf fmt "%a@." (print ~depth:0) (List.rev !log);
+    Format.fprintf fmt "%a@." (print ~depth:0) (Lazy.force l);
     close_out out_channel;
-    log := [];
-  end
+  end;
+  if !Clflags.inlining_report_serialized then begin
+    let ch = open_out_bin (output_prefix ^ ".inlining.obj.magic") in
+    let metadata = {
+      compilation_unit = Compilation_unit.get_current_exn ();
+    } in
+    let report : report = `Flambda2_1_0_0 (metadata, Lazy.force l) in
+    Marshal.to_channel ch report [];
+    close_out ch
+  end;
```

Comment:
I would put a try-with around this whole section; if any exception is raised, we can then print "Warning: inlining report writing failed: <exn>" or similar.

Reply:
Done (using `Misc.try_finally`). Is there a better way to print the warning than `eprintf`, considering that this is not a numbered warning and thus the `Warnings` module does not seem to help much ?

Reply:
Not that I know of, `eprintf` will be fine for now.